### ISSUE TITLE
[Design] - Profile: Update feedback alert pattern in direct deposit

### ISSUE THREAD

**Issue by andaleliz** [department-of-veterans-affairs/va.gov-team#41527]
We need to update the direct deposit feedback alerts to match the pattern in other sections of profile.

**Tasks**

- [ ] Determine if there are any other feedback alerts we need to account for other than success.  There might be one after multiple unsuccessful save attempts, like contact information.
- [ ] Get content review on any other feedback alerts discovered
- [ ] Update sketch file, desktop and mobile
- [ ] let PM know when ready for FE implementation

**Acceptance Criteria**

- feedback alert should appear directly above the button the user clicked to start the process.  E.g. success alert should appear above the edit button after a successful save
- success copy should read "Update saved"
- TBD on any other content, pending content review

**Comment by andaleliz** [comment]
@Samara-Strauss and @SKasimow I created this as a [follow up to the comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/39348#issuecomment-1129279184) in #39348. Put it in sprint 74 as a placeholder but please feel free to move based on other priorities.  Not sure if you want me or Christina on it so I added us both for now.

**Comment by andaleliz** [comment]
I've created the success update in our direct deposit Sketch files. There are some OS/Sketch version challenges ([Slack post](https://dsva.slack.com/archives/C01KY4VUN8P/p1653671518836789)) that I want to make sure @cgednalske doesn't struggle with while I'm out, so I worked on this locally and have not pushed this to the Sketch for Teams cloud yet. Once she is up and running w/ the latest OS and Sketch version, I'll merge that change into our existing DD file which will be viewable in a browser via a link.

@adamwhitlock1 or @mdewey do either of you know what the behavior is if there is a save failure in direct deposit, like if the API call fails mid-save, or we can't write to the db for some other reason? I know in contact information, if there are 3 failed save attempts, the user is returned to the read view and this error shows:
![image.png](https://images.zenhubusercontent.com/5d9def68b6de6d00018d50f2/1929561e-60c3-4c4f-bbca-ecffbf7e8643)
I'm trying to understand if there's something comparable in direct deposit we need to account for. I'm aware of the fraud alert and routing number errors shown here:
![image.png](https://images.zenhubusercontent.com/5d9def68b6de6d00018d50f2/6797d80a-5c19-4068-b2fa-111368c043ad)

@SKasimow I'm moving this into sprint 75 since I wasn't able to complete prior to my trip.

**Comment by mdewey** [comment]
So . 

All the error messages we have are located [here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/personalization/profile/components/direct-deposit/PaymentInformationEditError.jsx). That is the code, but the short answer is that we are showing errors for `FlaggedAccount`, `FlaggedRoutingNumber`, `InvalidRoutingNumber`, `UpdateAddressError`, `UpdateHomePhoneNumberError` and `UpdateWorkPhoneNumberError`.  I can get you the content if you need it. 

I do not see a re-tries max, but I am not 100% on that (more like 70%).

**Comment by adamwhitlock1** [comment]
To expand on what Mark said, the location of these PaymentInformationEditError instances will show [here](https://github.com/department-of-veterans-affairs/vets-website/blob/c9cbc2dd19bfbfd6c8b77b8bc51e3058370feb06/src/applications/personalization/profile/components/direct-deposit/BankInfo.jsx#L275) which aligns with the screenshots that you have captured in that it is _above_ the form fields in contrast to the the contact / personal information error alerts, which are _below_ the fields.

I also don't see any reference to the 3 attempts in either direct deposit or contact information actually. I do see that in the [ProfileInformationFieldController](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/user/profile/vap-svc/components/ProfileInformationFieldController.jsx#L101) there is a 5 second timeout for how long a transaction will be 'polled' for / retrying for the status of the transaction, and that the interval is 1 second, so that would equate to 5 'status' retries for things like contact information updates.

As far as a **generic error** for issues saving the payment information (api error or other saving problem), it looks like the content for that is specifically [here](https://github.com/department-of-veterans-affairs/vets-website/blob/c9cbc2dd19bfbfd6c8b77b8bc51e3058370feb06/src/applications/personalization/profile/components/direct-deposit/PaymentInformationEditError.jsx#L73), and it is used as a fallback if the errors that get returned don't match to any specific error. This generic error would visually look just like the other errors in that they would be in an AlertBox, and would display a headline of 'We couldn’t update your bank information'

**Comment by andaleliz** [comment]
Thanks @adamwhitlock1 and @mdewey, that info is super helpful.

I'll make my content recommendations + review with Danielle during Thursday's office hours.

**Comment by Samara-Strauss** [comment]
So there are two fraud flags:

- One is added to the account when there are legitimate concerns that someone's information has been compromised. This is added by the fraud team (separate team at the VA), and is removed once it is determined the veteran's info is safe. As far as I know, this is still active, and is what is illustrated in [this comment](https://github.com/department-of-veterans-affairs/va.gov-team/issues/41527#issuecomment-1140034510).
- The other fraud flag was for people who used a certain set of bank routing numbers. There were a set of *legitimate* routing numbers that were compromised in a fraud ring a few years ago, and the VA restricted people with those routing numbers so that they had to call instead of updating their info online as an additional security measure. However, I believe that these routing numbers are no longer restricted and that this error has been removed by the backend.

**Comment by Samara-Strauss** [comment]
@andaleliz I am not 100% sure it's been removed from the code. I am deducing that is has based on two things:

- In September 2020, the fraud team confirmed that this error was no longer needed. **However**, the backend team never responded to the email thread about removing this error, so I don't have actual confirmation from them that they removed this error state.
- That being said, I can see in analytics that this error was still coming in as of January 2021, but was no longer coming in by February 2021. I don't see any incidents of it after that point.

**Comment by andaleliz** [comment]
Got it - since we don't have official confirmation, it feels risky to remove it. Just in case. I will include this in the updates then.

**Comment by Samara-Strauss** [comment]
I can try to email BGS and see if they can confirm that this flag was removed. If I hear back, I'll let you know.

**Comment by andaleliz** [comment]
@Samara-Strauss sorry to keep bugging you with questions about this today. For the flagged routing number error, do you know if a user would be able to save changes to account number or type, or are they blocked from _any_ edits once this flag is in place?

Here's a bit more context - as I've looked more carefully at these alerts, I'm realizing we have a lot of opportunity to improve the UX. Currently, we have 3 (possibly 4) alerts that appear that essentially block the user from being able to make _any_ edits, in which case it doesn't make sense to keep them in edit mode (which we currently do). If we return them to read mode, it seems like it doesn't really even make sense for them to have an edit button anymore, because they'd just go back into a loop.

For the flagged routing number, if they _can_ edit account number or type, then it does make sense to display that error in edit vs read mode.

**Comment by andaleliz** [comment]
Created [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/42557) to review during content office hours tomorrow

**Comment by Samara-Strauss** [comment]
> As I've looked more carefully at these alerts, I've realized have a lot of opportunity to improve the UX. Currently, we have 3 (possibly 4) alerts that appear that essentially block the user from being able to make any edits. However, we display these alerts in edit mode, which doesn't make sense since they'd get the same exact error if they hit "update" again. We can return them to read mode and display the error there - and then I wonder if we should also remove the edit button entirely because they'd just go back into a loop if they tried again. 

In most cases, I think we can't know if there's an issue with a veteran updating their direct deposit until they update the form. So, I don't know that we'd be able to persist errors in read-only mode (like if they leave the page and come back), which means that (I think) there's no way to prevent people from submitting the form multiple times and getting the same error. You may have already done this, but it might help to sit down with Mark or Adam to talk about if there are better options for how we handle things now.

> For the flagged routing number error, do you know if a user would be able to save changes to account number or type, or are they blocked from any edits once this flag is in place?

Since you can't separate a routing number update from an account number update (since, effectively, they are are two parts of a whole), no, you could not update other information in the bank form and not update your routing number.

Also, I still haven't heard back from BGS as to whether they've officially removed this error, but I did email them. We likely won't hear back on this, but I wouldn't worry too much about this use case since it's most likely not a thing anymore. But, I still think we shouldn't remove the code just in case.

> For the flag added by the VA fraud team, do you know if the flag is added at the bank account level, or at the user level? I'm wondering if editing ability for both edu and comp payments would be blocked if they have different accounts for each direct deposit.

I think this is added at the user level, but given that the fraud team doesn't manage anything to do with edu, it might be possible to update one but not the other. But I'm not sure. I honestly don't remember the specifics of this use case at this point. We'd have to get in touch with the fraud team. I've reached out to my contact on the fraud team to see if he's still the right person to direct questions to or if we should reach out to someone else.

**Comment by Samara-Strauss** [comment]
OMG nevermind! We actually did hear back from BGS. I'll forward you the email as to keep details out of our public repo.

**Comment by Samara-Strauss** [comment]
I also sent an email to our fraud contacts with some questions. Feel free to reply to that email with additional questions. Keep in mind that they won't know anything about the VA.gov UI, but they can tell you generally how things are handled when fraud is suspected for comp & pen payments.

**Comment by andaleliz** [comment]
I met with Mark and Adam today to discuss alerts, and Danielle to discuss content.  We have 4 use cases in total to consider.

### Use cases
#### User entered an invalid routing number

This is a unique use case in that it references a specific field within a form and this particular validation happens after submit. Typically, field specific validation happens on input with our form system.  

I recommend we reconsider the UX for this alert. Currently, users see a giant error alert guiding them to a field that they have to search for. We significantly reduce cognitive load if we use an inline error message on the field, and jump (not scroll) users to this field when the error appears.  [Here's a mock-up](https://www.sketch.com/s/1a920e73-1dcb-47c4-aae8-08656756c131/v/n5K3pa/a/YGOQEkZ).  

Based on my convo with the devs, implementing this is likely going to require some customization on the form field and we'd need to do some technical discovery to estimate the work.  In the event that other initiatives will prevent that from happening soon, [I mocked up a version](https://www.sketch.com/s/1a920e73-1dcb-47c4-aae8-08656756c131/v/n5K3pa/a/ago74wq) that would not require customization and is still an improvement over what we have today.  This version uses a more mobile-friendly alert that's consistent with other user feedback alerts in profile, and includes an anchor link to guide the user to the field in question.  It strays from the pattern of displaying the feedback message directly above the update button (which we have in personal and contact info) but I think the uniqueness of this use case warrants that. We don't have any fields in profile that trigger a response from an API post-submit.

#### Flag added to the account for a routing number included in a fraud ring
We've determined this is no longer an issue. I recommend removing this from the FE code and Sketch files. At a minimum, comment it out of the code w/ a note about it no longer being used so a future version of our team doesn't have to repeat the work we've done.

#### Flag added to the account when there are legitimate concerns that someone's information has been compromised. This is the one added by the fraud team (separate team at the VA), and is removed once it is determined the veteran's info is safe.  We need to learn more about this one before I provide my recommendation and are currently discussing via email. I'll come back to this.

#### Invalid address or phone number
@Samara-Strauss the FE code contains two functions ([starting here](https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/applications/personalization/profile/components/direct-deposit/PaymentInformationEditError.jsx#L85)) to check the presence and validity of a person's mailing address and phone number. We tested the phone use case staging and are still able to make DD updates on staging if there are no phone numbers present.  Some questions for you:
1. Do you know if people do indeed need to have a mailing address and phone number in their profile to update bank information? If so:
2. Is it checking for a home or mobile number?
3. Are the DD back ends on staging running the appropriate checks? E.g. is it possible this can only be reproduced in production?

**Comment by Samara-Strauss** [comment]
**User entered an invalid routing number**

Given that *I think* the form has to be submitted in order to determine that the routing number is invalid, I recommend going with the version that requires less customization. 

**Flag added to the account for a routing number included in a fraud ring**

I'm OK with removing this since the backend team confirmed that they are no longer using this flag.

**Invalid address or phone number**

> Do you know if people do indeed need to have a mailing address and phone number in their profile to update bank information? If so:

I don't know. I forget if this was an issue with needing to have *something* on file, or if it was an issue with potential inconsistencies in records (eg. different home phone numbers in different databases). We'd honestly need to ask EVSS. It's possible we explained this thoroughly in the contact center guide as well, so that's also a place to look for answers. 

> Is it checking for a home or mobile number?

It is checking for home and work numbers. Gonna keep analytics out of the public repo, but sent you a screenshot of what I see in GA over email.

> Are the DD back ends on staging running the appropriate checks? E.g. is it possible this can only be reproduced in production?

I would assume so, but I don't know that any staging users meet this use case. This was probably something we had to implement with no real way to test for it.

**Comment by andaleliz** [comment]
>Given that I think the form has to be submitted in order to determine that the routing number is invalid, I recommend going with the version that requires less customization.

Sounds good. 

In that case, @SKasimow, that piece is ready for a FE implementation ticket. I [started a ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/42734) but didn't add any labels or assignees.

You can also create a FE ticket to tackle the work for removing/commenting out the code for the use case "**Flag added to the account for a routing number included in a fraud ring**"

>I don't know. I forget if this was an issue with needing to have something on file, or if it was an issue with potential inconsistencies in records (eg. different home phone numbers in different databases). We'd honestly need to ask EVSS. It's possible we explained this thoroughly in the contact center guide as well, so that's also a place to look for answers.

I found some documentation in the call center guide that was helpful. I'm wondering how we might provide more clear information up front about what is required so that people don't encounter these errors in the first place. I can sync with EVSS on these questions. I'll make a ticket for this discovery work.

- Does the backend check for home, work, then address, in that order? The call center guide suggests that.
- What is driving this requirement for users to have these items valid in their profile?
- Does this apply to both kinds of direct deposit payments that we support?

>Flag added to the account when there are legitimate concerns that someone's information has been compromised.

We're meeting with this team on Wednesday to get answers to our questions.  I'm going to close this ticket for now, and spin up a different one to cover that.

**Comment by Samara-Strauss** [comment]
>We're meeting with this team on Wednesday to get answers to our questions. 

The fraud team will only be able to answer questions about what happens when they flag someone's account as a potential victim of fraud. They are not EVSS, and they won't be able to answer your questions about the address and phone number checks. I would suggest asking in the #evss-dev room in Slack who there is an expert on the comp & pen direct deposit integration. It used to be Mike Walker, but he left recently.
